core/src/sumeragi/main_loop.rs

```diff
@@ -0,0 +1,1340 @@
+                .sorted_peers()
+                .iter()
+                .cloned()
+                .collect(),
+            state_machine.current_topology.max_faults(),
+            &state_machine.latest_block_hash,
+            &suspect_proof,
+        )
+        .unwrap_or_else(|err| error!("{err}"));
+
+    sumeragi.broadcast_packet(
+        MessagePacket::new(
+            view_change_proof_chain.clone(),
+            Message::ViewChangeSuggested,
+        ),
+        &state_machine.current_topology,
+    );
+}
+
+fn prune_view_change_proofs_and_calculate_current_index(
+    state_machine: &State,
+    view_change_proof_chain: &mut Vec<Proof>,
+) -> u64 {
+    view_change_proof_chain.prune(&state_machine.latest_block_hash);
+    view_change_proof_chain.verify_with_state(
+        &state_machine
+            .current_topology
+            .sorted_peers()
+            .iter()
+            .cloned()
+            .collect(),
+        state_machine.current_topology.max_faults(),
+        &state_machine.latest_block_hash,
+    ) as u64
+}
+
+fn handle_role_agnostic_messages<F>(
+    sumeragi: &SumeragiWithFault<F>,
+    state_machine: &mut State,
+    maybe_incoming_message: &mut Option<Message>,
+) where
+    F: FaultInjection,
+{
+    if let Some(stolen_message) = maybe_incoming_message.take() {
+        match stolen_message {
+            Message::TransactionGossip(tx_gossip) => {
+                for transaction in tx_gossip.txs {
+                    let tx_maybe = VersionedAcceptedTransaction::from_transaction(
+                        transaction.into_v1(),
+                        &sumeragi.transaction_limits,
+                    );
+                    if let Ok(tx) = tx_maybe {
+                        match sumeragi.queue.push(tx, &state_machine.wsv) {
+                            Err((_, crate::queue::Error::InBlockchain)) | Ok(()) => {}
+                            Err((_, err)) => {
+                                warn!(?err, "Failed to push to queue gossiped transaction.")
+                            }
+                        }
+                    }
+                }
+            }
+            Message::ViewChangeSuggested => {
+                trace!("Received view change suggestion.");
+            }
+            other => *maybe_incoming_message = Some(other),
+        }
+    }
+}
+
+#[allow(clippy::too_many_arguments)]
+fn compare_view_change_index_and_block_height_to_old<F>(
+    sumeragi: &SumeragiWithFault<F>,
+    current_view_change_index: u64,
+    old_view_change_index: &mut u64,
+    current_latest_block_height: u64,
+    old_latest_block_height: &mut u64,
+    // below is the state that gets reset.
+    current_topology: &mut Topology,
+    voting_block_option: &mut Option<VotingBlock>,
+    block_signature_acc: &mut Vec<(
+        HashOf<VersionedValidBlock>,
+        SignatureOf<VersionedValidBlock>,
+    )>,
+    has_sent_transactions: &mut bool,
+    instant_when_we_should_create_a_block: &mut Instant,
```

Comment:
😧

Reply:
Okay, I understand, that it's just the best we can do so simplify the main loop for now.

But I don't think it's okay for the future

Reply:
So, there's two things, on the one hand, you kinda have to concede that the new approach while not conducive to modification (which should happen rarely), is conducive to debugging (which might happen often, since we need to rule out consensus first in most cases). 

Normally I wouldn't have accepted this change at all, but we can work our way around it. Given that I or @SamHSmith have no problem navigating the main loop and no-one else is supposed to anyway, I'm all for earmarking a refactor, but doing it in October/November.